addons/mass_mailing/models/mail_mail.py

```diff
@@ -86,29 +77,31 @@ def _sub_relative2absolute(match):
         body = re.sub('(<img(?=\s)[^>]*\ssrc=")(/[^/][^"]+)"', _sub_relative2absolute, body)
 
         # generate tracking URL
-        if mail.statistics_ids:
-            tracking_url = self._get_tracking_url(cr, uid, mail, partner, context=context)
+        if self.statistics_ids:
+            tracking_url = self._get_tracking_url(partner)
             if tracking_url:
                 body = tools.append_content_to_html(body, tracking_url, plaintext=False, container_tag='div')
         return body
 
-    def send_get_email_dict(self, cr, uid, ids, partner=None, context=None):
+    @api.multi
+    def send_get_email_dict(self, partner=None):
         # TDE: temporary addition (mail was parameter) due to semi-new-API
-        res = super(MailMail, self).send_get_email_dict(cr, uid, ids, partner, context=context)
-        mail = self.browse(cr, uid, ids[0], context=context)
-        base_url = self.pool.get('ir.config_parameter').get_param(cr, uid, 'web.base.url')
-        if mail.mailing_id and res.get('body') and res.get('email_to'):
+        self.ensure_one()
+        res = super(MailMail, self).send_get_email_dict(partner)
+        base_url = self.env['ir.config_parameter'].get_param('web.base.url')
+        if self.mailing_id and res.get('body') and res.get('email_to'):
             emails = tools.email_split(res.get('email_to')[0])
             email_to = emails and emails[0] or False
-            unsubscribe_url= self._get_unsubscribe_url(cr, uid, mail, email_to, context=context)
-            link_to_replace =  base_url+'/unsubscribe_from_list'
+            unsubscribe_url = self._get_unsubscribe_url(email_to)
+            link_to_replace = base_url + '/unsubscribe_from_list'
             if link_to_replace in res['body']:
                 res['body'] = res['body'].replace(link_to_replace, unsubscribe_url if unsubscribe_url else '#')
         return res
 
-    def _postprocess_sent_message(self, cr, uid, mail, context=None, mail_sent=True):
+    @api.model
+    def _postprocess_sent_message(self, mail, mail_sent=True):
```

Comment:
can you check this method "_postprocess_sent_message_v9" ? all code of "_postprocess_sent_message" method is shifted in this method "_postprocess_sent_message_v9" and keep old method for old api..